tests/models.py

```diff
@@ -224,3 +224,37 @@ class Meta:
     #       makes some kind of sense for the question type
     statement = factory.LazyFunction(_required)
     type = factory.LazyFunction(_required)
+
+
+class _InvitationFactory(factory.alchemy.SQLAlchemyModelFactory):
+    class Meta:
+        model = Invitation
+        sqlalchemy_session_factory = lambda: db.session  # noqa: E731
+        sqlalchemy_session_persistence = "commit"
+
+    id = factory.LazyFunction(uuid4)
+    email = factory.Faker("email")
+    user_id = None
+    user = None
+    organisation_id = None
+    organisation = None
+    grant_id = None
+    grant = None
+    role = None
+    expires_at_utc = factory.LazyFunction(lambda: datetime.datetime.now() + datetime.timedelta(days=7))
+    claimed_at_utc = None
+
+    class Params:
+        has_organisation = factory.Trait(
+            organisation_id=factory.LazyAttribute(lambda o: o.organisation.id),
+            organisation=factory.SubFactory(_OrganisationFactory),
+        )
+        has_grant = factory.Trait(
+            grant_id=factory.LazyAttribute(lambda o: o.grant.id),
+            grant=factory.SubFactory(_GrantFactory),
+        )
+        is_claimed = factory.Trait(
+            claimed_at_utc=factory.LazyFunction(lambda: datetime.datetime.now()),
+            user=factory.SubFactory(_UserFactory),
+            user_id=factory.LazyAttribute(lambda o: o.user.id if o.user else None),
```

Comment:
These traits feel like really nice ways to encapsulate the logic and assist with setup - good idea 🙇